devito/operations/interpolators.py

```diff
@@ -235,10 +305,25 @@ def _interp_idx(self, variables, implicit_dims=None, pos_only=()):
         temps.extend(self._coeff_temps(implicit_dims))
 
         # Substitution mapper for variables
-        mapper = self._rdim._getters
-        idx_subs = {v: v.subs({k: c + p
-                    for ((k, c), p) in zip(mapper.items(), pos)})
-                    for v in variables}
+        mapper = self._rdim._getters.copy()
+
+        # Subsitution from SubDimensions to ConditionalDimensions
+        smapper = {}
+        # Substitution from ConditionalDimensions to rebuilt dimensions
+        cdmapper = {}
+        # Need to adjust bounds if Function defined on a SubDomain
+        if subdomain:
+            adjust_interp_indices(subdomain, mapper, smapper, cdmapper)
```

Comment:
This seems overlly intricated and overkill to build everything then redo a pass to replace everything with the input dimensions. What's the problem with doing `self._rrdims(grid)` and `self._gdims(grid)` and catching directly the right dimensions instead of rebuilding everything?

Reply:
Yeah I was thinking something along those lines. I figured it would be best to get it working then refactor it